diff --git a/perlbrew/Dockerfile b/perlbrew/Dockerfile
index cd7b03a..3b36b56 100644
--- a/perlbrew/Dockerfile
+++ b/perlbrew/Dockerfile
@@ -1,14 +1,15 @@
 # build a perlbrew enabled image
 # example: docker build -t perlbrew:stable .
 
-FROM ubuntu:bionic
+FROM debian:buster
 
 MAINTAINER "Michael Fung <hkuser2001@gmail.com>"
 
 ENV DEBIAN_FRONTEND noninteractive
 
 ENV HOME /root
-
+ENV PERLBREW_ROOT /opt/perlbrew
 RUN apt-get update \
     && apt-get -y install perlbrew \
+    && \curl -L https://install.perlbrew.pl | bash \
     && apt-get clean && rm -rf /var/lib/apt/lists/*  # cleanup to save space
diff --git a/rt-builder/Dockerfile b/rt-builder/Dockerfile
index 9da44a5..acdc964 100644
--- a/rt-builder/Dockerfile
+++ b/rt-builder/Dockerfile
@@ -25,7 +25,7 @@ RUN apt-get update \
     && apt-get install -y git curl build-essential \
     && apt-get install -y libev4 libev-dev libffi6 libffi-dev \
     && apt-get install -y libzmq5 libzmq3-dev \
-    && apt-get install -y openssl libssl1.0.0 libssl1.1 libssl-dev libnss3 libnss3-dev \
+    && apt-get install -y openssl libssl1.1 libssl-dev libnss3 libnss3-dev \
     && apt-get install -y zlib1g zlib1g-dev \
     && apt-get clean && rm -rf /var/lib/apt/lists/*  # cleanup to save space
 
@@ -35,11 +35,12 @@ RUN apt-get update \
 ARG PERL_CPANM_OPT
 ENV PERLBREW_ROOT /opt/perlbrew
 COPY modules-list /
-RUN source /opt/perlbrew/etc/bashrc \
-    && perlbrew use 5.22.0-1.001 \
-    && xargs < modules-list | cpanm -M http://localcpan.lan:3111 --no-man-pages \
-    && rm -rf /opt/perlbrew/build \
-    && rm -f /opt/perlbrew/*log
+
+#RUN source /opt/perlbrew/etc/bashrc \
+#    && perlbrew use 5.22.0-1.001 \
+#    && xargs < modules-list | cpanm --no-man-pages \
+#    && rm -rf /opt/perlbrew/build \
+#    && rm -f /opt/perlbrew/*log
 
 #
 # setup docker run entry
diff --git a/stableperl-rt/Dockerfile b/stableperl-rt/Dockerfile
index 636c7f7..2bbbf6a 100644
--- a/stableperl-rt/Dockerfile
+++ b/stableperl-rt/Dockerfile
@@ -1,9 +1,9 @@
 # build the runtime image
-# example: docker build -t rt:stable .
+# example: docker build -t stableperl-rt:stable .
 
 FROM rt-builder:stable as builder
 
-FROM ubuntu:bionic
+FROM debian:buster
 
 MAINTAINER "Michael Fung <hkuser2001@gmail.com>"
 ENV DEBIAN_FRONTEND noninteractive
@@ -17,9 +17,10 @@ COPY 01_nodoc /etc/dpkg/dpkg.cfg.d/01_nodoc
 # install required binary libs
 #
 RUN apt-get update \
+    && apt-get install -y curl ca-certificates less procps lsof \
     && apt-get install -y libev4 libffi6 \
-    && apt-get install -y libzmq5 \
-    && apt-get install -y openssl libssl1.0.0 libssl1.1 libnss3 \
+    && apt-get install -y libzmq5 libzmq3-dev \
+    && apt-get install -y openssl libssl1.1 libnss3 \
     && apt-get install -y zlib1g \
     && apt-get clean && rm -rf /var/lib/apt/lists/*  # cleanup to save space
 
diff --git a/stableperl/Dockerfile b/stableperl/Dockerfile
index b9385d7..faebbf2 100644
--- a/stableperl/Dockerfile
+++ b/stableperl/Dockerfile
@@ -25,8 +25,8 @@ SHELL ["/bin/bash", "-c"]
 ENV PERLBREW_ROOT /opt/perlbrew
 RUN apt-get update \
     && apt-get -y install git curl build-essential \
-    && perlbrew init \
-    && mkdir -p $PERLBREW_ROOT \
+    && source /opt/perlbrew/etc/bashrc \
+    && perlbrew --version \
     && perlbrew install -j 4 --64int /stableperl-5.22.0-1.001.tar.gz \
     && rm -f stableperl-5.22.0-1.001.tar.gz \
     && apt-get clean && rm -rf /var/lib/apt/lists/*  # cleanup to save space
